addons/purchase/models/purchase.py

```diff
@@ -985,7 +991,7 @@ def make_po(self):
             if not suppliers:
                 procurement.message_post(body=_('No vendor associated to product %s. Please set one to fix this procurement.') % (procurement.product_id.name))
                 continue
-            supplier = suppliers[0]
+            supplier = self._make_po_select_supplier(suppliers)
```

Comment:
You're right, as this way you can take another criteria from the origin procurement.